src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -210,58 +223,60 @@ public LocalRepository localRepositoryFor(GitHubProject project) throws IOExcept
   private LocalRepository loadLocalRepositoryFor(GitHubProject project) throws IOException {
     Objects.requireNonNull(project, "On no! Project is null!");
 
-    LocalRepositoryInfo info = localRepositoriesInfo.get(project.url());
-    if (info == null) {
-      Path repositoryPath = base
-          .resolve(project.organization().name())
-          .resolve(project.name());
+    synchronized (LOCAL_REPOSITORIES_LOCK) {
+      LocalRepositoryInfo info = localRepositoriesInfo.get(project.url());
```

Comment:
Strictly speaking, `localRepositoriesInfo` should be static. But let's leave it as-is for now. We'll consider this update later.